test/src/test/java/org/corfudb/runtime/collections/DiskBackedCorfuClientTest.java

```diff
@@ -0,0 +1,171 @@
+        }
+
+        @Override
+        public byte getType() {
+            return 23; // Random number.
+        }
+
+        @Override
+        public Object deserialize(ByteBuf b, CorfuRuntime rt) {
+            return gson.fromJson(new String(RocksDbStreamingMap.byteArrayFromBuf(b)), clazz);
+        }
+
+        @Override
+        public void serialize(Object o, ByteBuf b) {
+            b.writeBytes(gson.toJson(o).getBytes());
+        }
+    }
+
+    @Data
+    @Builder
+    public static class Pojo {
+        public final String payload;
+    }
+
+    @Test
+    public void customSerializer() throws RocksDBException {
+        RocksDB.loadLibrary();
+
+        final File persistedCacheLocation = new File("/tmp/", "diskBackedMap");
+
+        final Options options =
+                new Options().setCreateIfMissing(true)
+                        // The size is checked either during flush or compaction.
+                        .setWriteBufferSize(FileUtils.ONE_KB);
+
+        CorfuTable<String, Pojo>
+                table = getDefaultRuntime().getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<String, Pojo>>() {})
+                .setArguments(new RocksDbStreamingMap<String, Pojo>(
+                        persistedCacheLocation, options,
+                        new PojoSerializer(Pojo.class), getRuntime()))
+                .setStreamName("diskBackedMap")
+                .open();
+
+        final long ITERATION_COUNT = 100;
+        final int ENTITY_CHAR_SIZE = 100;
+
+        LongStream.rangeClosed(1, ITERATION_COUNT).forEach(idx -> {
+            String key = RandomStringUtils.random(ENTITY_CHAR_SIZE, true, true);
+            Pojo value = Pojo.builder()
+                    .payload(RandomStringUtils.random(ENTITY_CHAR_SIZE, true, true))
+                    .build();
+            table.put(key, value);
+            Pojo persistedValue = table.get(key);
+            Assertions.assertEquals(value, persistedValue);
+        });
+    }
+
+    @Test
+    public void fileSystemLimit() throws RocksDBException {
+        RocksDB.loadLibrary();
+
+        final File persistedCacheLocation = new File("/tmp/", "diskBackedMap");
+        SstFileManager sstFileManager = new SstFileManager(Env.getDefault());
+        sstFileManager.setMaxAllowedSpaceUsage(FileUtils.ONE_KB);
+        sstFileManager.setCompactionBufferSize(FileUtils.ONE_KB);
+
+        final Options options =
+                new Options().setCreateIfMissing(true)
+                        .setSstFileManager(sstFileManager)
+                        // The size is checked either during flush or compaction.
+                        .setWriteBufferSize(FileUtils.ONE_KB);
+
+        CorfuTable<String, String>
+                table = getDefaultRuntime().getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<String, String>>() {})
+                .setArguments(new RocksDbStreamingMap<String, String>(
```

Comment:
I think we should move the naming away from RocksDB. If we change the technology or have a different implementation in the future, all clients will have to change this. Maybe PersistedStreamingMap or something like that.

Reply:
Done.

Remember, Corfu clients will not be working with the `CorfuTable` directly, but rather via the `CorfuStore`, so this implementation detail will not leak outside of Corfu.

I thought about abstracting `RocksDbStreamingMap` away, but the problem is that each of the options is very specific to the embedded database we end up using (take a look at the `org.rocksdb.Options` parameter).